image/templates/helm/stackrox-central/templates/01-central-13-deployment.yaml

```diff
@@ -100,7 +100,7 @@ spec:
           value: {{ ._rox.central.disableTelemetry | not | quote }}
         - name: ROX_OFFLINE_MODE
           value: {{ ._rox.env.offlineMode | quote }}
-        {{- if eq ._rox.env.openshift 4 }}
+        {{- if and (eq ._rox.env.openshift 4) (not (hasKey ._rox.customize.central.envVars "ROX_MANAGED_CENTRAL" )) }}
```

Comment:
>How do we know that ROX_MANAGED_CENTRAL is always already set here?

I'm not sure I understand your concern here. The `rox.customize.central.envVars` is supposed to be set within the `values.yaml` of the Helm Chart. At this point, of rendering the template, we can be sure that its either set / or not set, due to the `values.yaml` being used as input for rendering. So I'm not sure about your concern.

I get that it's weird that we set a env var based on the value of another env var, but what would be the alternative? We don't have anything besides `ROX_MANAGED_CENTRAL` that signals whether we are running in managed service context or not. Also, we already do set the env variable based on another env variable: `env.openshift=4`.

Reply:
But why do we need to set `ROX_ENABLE_OPENSHIFT_AUTH` to `false`, if there is already this check:

```
if env.EnableOpenShiftAuth.BooleanSetting() && !env.ManagedCentral.BooleanSetting()
```

I think I would prefer enforcement on the code level rather than the deployment level. That way it will work for all deployment methods.

Reply:
Yeah, I can remove it for the deployment template, but we will have to keep it within the service account template.